lib/pub_sub/message.rb

```diff
@@ -11,20 +11,22 @@ def initialize(payload)
     end
 
     def process
-      validate_message!
+      begin
+        validate_message!
+      rescue PubSub::ServiceUnknown => e
+        return
+      end
       handler.process(data)
     end
 
     def validate_message!
       messages = PubSub.config.subscriptions[sender]
-      if messages.nil?
-        warning = "WARN: We received a message from #{sender} but we do " \
-                'not subscribe to that service.'
-        PubSub.logger.warn(warning)
-      end
-      unless messages.include?(type)
-        error = "We received a message from #{sender} but it was " \
-                "of unknown type #{type}."
+      if messages.blank?
+        warning = "#{PubSub.config.service_name} received a message from #{sender} but no matching subscription exists for that sender"
```

Comment:
I don't think these wrapped lines should be reformatted to a single line.  This is how it currently looks on my screen in portrait orientation:

![screen shot 2015-09-17 at 10 40 22 am](https://cloud.githubusercontent.com/assets/279925/9922152/912aaf78-5d28-11e5-844b-cc035107d1cf.png)

And this is how it looks after your reformat. Notice how the lines extend past the edge of my screen and the wrapping makes it difficult to read:

![screen shot 2015-09-17 at 10 39 55 am](https://cloud.githubusercontent.com/assets/279925/9922155/9423e4c4-5d28-11e5-87d2-44df646d6b01.png)